lib/ui/painting.dart

```diff
@@ -4960,6 +4960,22 @@ class ImmutableBuffer extends NativeFieldWrapperClass2 {
   }
   void _init(Uint8List list, _Callback<void> callback) native 'ImmutableBuffer_init';
 
+  /// Create a buffer from the asset with key [assetKey].
+  ///
+  /// Returns `null` if the asset does not exist.
+  static ImmutableBuffer? fromAsset(String assetKey) {
+    final ImmutableBuffer buffer = ImmutableBuffer._(0);
+    bool success = false;
+    _fromAsset(assetKey, buffer, (void result) {
+      success = true;
+    });
+    if (!success) {
+      return null;
+    }
+    return buffer;
+  }
+  static void _fromAsset(String assetKey, ImmutableBuffer buffer, _Callback<void> callback) native 'ImmutableBuffer_fromAsset';
```

Comment:
The more typical signature for this is `String? _fromAsset(ImmutableBuffer outBuffer, String assetKey, _Callback<void> callback) native 'ImmutableBuffer_fromAsset;`. The String is used to return an error message rather than throwing in C++, which has undesirable runtime overhead/stack trace strangeness. You can then use `_futureize` with it.

It doesn't need to be static, and if the callback calls back with an int it could return the length.